synapse/models/planning.py

```diff
@@ -0,0 +1,141 @@
+import synapse.lib.module as s_module
+
+class PlanModule(s_module.CoreModule):
+
+    def getModelDefs(self):
+        return (('plan', {
+            'types': (
+                ('plan:system', ('guid', {}), {
+                    'doc': 'A planning or behavioral analysis system that defines phases and procedures.'}),
+
+                ('plan:phase', ('guid', {}), {
+                    'doc': 'A phase within a planning system which may be used to group steps.'}),
+
+                ('plan:procedure', ('guid', {}), {
+                    'doc': 'A procedure consisting of steps.'}),
+
+                ('plan:procedure:type:taxonomy', ('taxonomy', {}), {
+                    'interfaces': ('meta:taxonomy',),
+                    'doc': 'A taxonomy of procedure types.'}),
+
+                ('plan:procedure:variable', ('guid', {}), {
+                    'doc': 'A variable used as input/output by a procedure.'}),
+
+                ('plan:procedure:step', ('guid', {}), {
+                    'doc': 'A step within a procedure.'}),
+
+                ('plan:procedure:link', ('guid', {}), {
+                    'doc': 'A link between steps in a procedure.'}),
+            ),
+
+            'forms': (
+                ('plan:system', {}, (
+                    ('name', ('str', {'lower': True, 'onespace': True}), {
+                        'ex': 'mitre att&ck flow'
+                        'doc': 'The name of the planning system.'}),
+                    ('author' ('ps:contact', {}), {
+                        'doc': 'The contact of the authoring person or organization.'}),
+                    ('created', ('time', {}), {
+                        'doc': 'The time the planning system was first created.'}),
+                    ('url', ('inet:url', {}), {
+                        'doc': 'The primary URL which documents the planning system.'}),
+                )),
+                ('plan:phase', {}, (
+                    ('title', ('str', {}), {
+                        'ex': 'Reconisence Phase'
+                        'doc': 'The title of the phase.'}),
+
+                    ('summary', ('str', {}), {
+                        'doc': 'A summary of the definition of the phase.'}),
+
+                    ('index', ('int', {}), {
```

Comment:
When we use index here, are we using it to note that this is phase 3 out of 4, etc?

Reply:
Yep! It's there to provide an arbitrary sorting order.